arkane/input.py

```diff
@@ -214,12 +214,11 @@ def species(label, *args, **kwargs):
         else:
             if structure:
                 # If a structure was given, simply calling spec.molecularWeight will calculate the molecular weight
-                spec.molecularWeight = spec.molecularWeight
-            elif is_pdep(jobList):
-                # If one of the jobs is pdep and no molecular weight is given or calculated, raise an error
-                raise ValueError("No molecularWeight was entered for species {0}. Since a structure wasn't given"
-                                 " as well, the molecularWeight, which is important for pressure dependent jobs,"
-                                 " cannot be reconstructed.".format(spec.label))
+                if spec.molecularWeight is None and is_pdep(jobList):
+                    # If one of the jobs is pdep and no molecular weight is given or calculated, raise an error
+                    raise ValueError("No molecularWeight was entered for species {0}. Since a structure wasn't given"
```

Comment:
Hmm, my suggestion might not have been correct, since this now misses the case where a structure is not provided. One possibility:
```
if molecularWeight is not None:
    spec.molecularWeight = molecularWeight
elif spec.molecularWeight is None and is_pdep(jobList):
    raise ValueError()
```
There shouldn't be any need to check whether structure exists, because the the Species.molecularWeight property checks if it has a molecule attribute, which will be empty if structure was empty.